packages/amplify_datastore/example/integration_test/query_test/query_predicate_test/compound_query_predicate_test.dart

```diff
@@ -0,0 +1,375 @@
+                .and(MultiTypeModel.ALTINTVALUE.ne(valueThree)),
+          );
+          expect(ListEquality().equals(actualModels, expectedModels), isTrue);
+        });
+
+        testWidgets('ne() & ne() on the same field',
+            (WidgetTester tester) async {
+          var expectedModels = models
+              .where((model) => model.intValue != valueTwo)
+              .where((model) => model.intValue != valueThree)
+              .toList();
+          var actualModels = await Amplify.DataStore.query(
+            MultiTypeModel.classType,
+            where: MultiTypeModel.INTVALUE
+                .ne(valueTwo)
+                .and(MultiTypeModel.INTVALUE.ne(valueThree)),
+          );
+          expect(ListEquality().equals(actualModels, expectedModels), isTrue);
+        });
+
+        testWidgets('lt() & lt()', (WidgetTester tester) async {
+          var expectedModels = models
+              .where((model) => model.intValue!.compareTo(valueTwo) < 0)
+              .where((model) => model.altIntValue!.compareTo(valueThree) < 0)
+              .toList();
+          var actualModels = await Amplify.DataStore.query(
+            MultiTypeModel.classType,
+            where: MultiTypeModel.INTVALUE
+                .lt(valueTwo)
+                .and(MultiTypeModel.ALTINTVALUE.lt(valueThree)),
+          );
+          expect(ListEquality().equals(actualModels, expectedModels), isTrue);
+        });
+
+        testWidgets('gt() & gt()', (WidgetTester tester) async {
+          var expectedModels = models
+              .where((model) => model.intValue!.compareTo(valueOne) > 0)
+              .where((model) => model.altIntValue!.compareTo(valueTwo) > 0)
+              .toList();
+          var actualModels = await Amplify.DataStore.query(
+            MultiTypeModel.classType,
+            where: MultiTypeModel.INTVALUE
+                .gt(valueOne)
+                .and(MultiTypeModel.ALTINTVALUE.gt(valueTwo)),
+          );
+          expect(ListEquality().equals(actualModels, expectedModels), isTrue);
+        });
+      });
+
+      group('multiple types', () {
+        var stringValueOne = 'abc';
+        var stringValueTwo = 'xyz';
+        var intValueOne = 1;
+        var intValueTwo = 2;
+
+        // generate a model with every combination
+        var models = List.generate(
+          16,
+          (index) => MultiTypeModel(
+            intValue: index % 2 < 1 ? intValueOne : intValueTwo,
+            altIntValue: index % 4 < 2 ? intValueOne : intValueTwo,
+            stringValue: index % 8 < 4 ? stringValueOne : stringValueTwo,
+            altStringValue: index % 16 < 8 ? stringValueOne : stringValueTwo,
+          ),
+        );
+
+        setUpAll(() async {
+          await configureDataStore();
+          await clearDataStore();
+          for (var model in models) {
+            await Amplify.DataStore.save(model);
+          }
+        });
+
+        testWidgets('eq() (and)', (WidgetTester tester) async {
```

Comment:
nit: should it be `eq() & eq()` following other naming pattern above? Ditto below test descriptions

Reply:
Since it is testing 4 predicates together, it would be `eq() & eq() & eq() & eq()` following the pattern, but I thought that was pretty verbose